drivers/serial/uart_stm32.c

```diff
@@ -53,6 +53,12 @@ static inline void uart_stm32_set_baudrate(struct device *dev, u32_t baud_rate)
 		return;
 	}
 
+#if defined(CONFIG_LPUART_1) \
+	&& defined(CONFIG_SOC_SERIES_STM32G4X) \
+	&& defined(CONFIG_UART_CONSOLE)
```

Comment:
I wonder why the dependency on `CONFIG_UART_CONSOLE`.

Reply:
this fix will output correctly on console on nucleo_g431rb (where console is assigned to LPUART1). Other use cases are not considered (tested).